lib/rules/no-useless-constructor.js

```diff
@@ -25,13 +25,20 @@ module.exports = function(context) {
             return false;
         }
 
-
         return body[0].expression.arguments.every(function(arg, index) {
-            return (arg.type === "Identifier" && arg.name === ctorParams[index].name) ||
```

Comment:
This works, but I would write it more like

```js
var superArgs = body[0].expression.arguments;
var isSimpleParameterList = ctorParams.every(function(param, index) {
    return param.type === "Identifier" || param.type === "RestElement";
});
var spreadsArguments = 
    superArgs.length === 1 &&
    superArgs[0].type === "SpreadElement" &&
    superArgs[0].argument.name === "arguments";
var passesParamsAsArgs =
    superArgs.length === ctorParams.length && 
    superArgs.every(function(arg, index) {
        return (
            arg.type === "Identifier" &&
            arg.name === ctorParams[index].name
        ) || (
            arg.type === "SpreadElement" &&
            ctorParams[index].type === "RestElement" &&
            arg.argument.name === ctorParams[index].argument.name
        );
    });
return isSimpleParameterList && spreadsArguments || passesParamsAsArgs;
```

Also, I do some additional length checks which are necessary.